### ISSUE TITLE
error: unable to get the Kubernetes InClusterConfig: open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory

### ISSUE THREAD

**Issue by SomaticIT** [telepresenceio/telepresence#2876]
**Describe the bug**
When trying to intercept some service, I get the following error on the pod:
`error: unable to get the Kubernetes InClusterConfig: open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory`

Indeed, my deployment is created using `automountServiceAccountToken: false`.
Is there a way to avoid using the mounted service account token?

**To Reproduce**
1. When I run `telepresence intercept my-service --namespace my-ns --port 3000 -- npm run dev`
2. I see `telepresence: error: Back-off restarting failed container`
3. So I look at pod logs
4. I see `error: unable to get the Kubernetes InClusterConfig: open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory`
5. See attached logs

**Expected behavior**
The intercept should work :)

**Versions (please complete the following information):**
 - Output of `telepresence version`: 
 ```
Client: v2.8.5 (api v3)
Root Daemon: v2.8.5 (api v3)
User Daemon: v2.8.5 (api v3)
```

 - Operating system of workstation running `telepresence` commands
`Ubuntu 22.10 (WSL2)` 

 - Kubernetes environment and Version [e.g. Minikube, bare metal, Google Kubernetes Engine]
 `kubernetes cluster v1.23.0`

**Additional context**
Deployment is created using `automountServiceAccountToken: false`.

[telepresence_logs.zip](https://github.com/telepresenceio/telepresence/files/9973828/telepresence_logs.zip)

**Comment by wimift** [comment]
I have the same problem

**Comment by Hatament** [comment]
I have the same problem

**Comment by thallgren** [comment]
Thanks for reporting this. I've analyzed the problem and a fix will be available in the next release of the traffic agent.

**Comment by SomaticIT** [comment]
Good news, thank you!

**Comment by thallgren** [comment]
A fix for this was released with Telepresence 2.9.3. The fix is in the enhanced traffic-agent version 1.13.4 and the traffic-manager should discover that automatically.

**Comment by SomaticIT** [comment]
Perfect, we will try this, thank you!
